tests/test_general.py

```diff
@@ -27,899 +26,941 @@
+    assert fit_result.value(sig) / np.std(xdata) == pytest.approx(1, 1e-6)
+
+
+def test_likelihood_fitting_bivariate_gaussian():
+    """
+    Fit using the likelihood method.
+    """
+    # Make variables and parameters
+    x = Variable('x')
+    y = Variable('y')
+    x0 = Parameter('x0', value=0.6, min=0.5, max=0.7)
+    sig_x = Parameter('sig_x', value=0.1, max=1.0)
+    y0 = Parameter('y0', value=0.7, min=0.6, max=0.9)
+    sig_y = Parameter('sig_y', value=0.05, max=1.0)
+    rho = Parameter('rho', value=0.001, min=-1, max=1)
+
+    pdf = BivariateGaussian(x=x, mu_x=x0, sig_x=sig_x, y=y, mu_y=y0,
+                            sig_y=sig_y, rho=rho)
+
+    # Draw 100000 samples from a bivariate distribution
+    mean = [0.59, 0.8]
+    r = 0.6
+    cov = np.array([[0.11 ** 2, 0.11 * 0.23 * r],
+                    [0.11 * 0.23 * r, 0.23 ** 2]])
+    np.random.seed(42)
+    xdata, ydata = np.random.multivariate_normal(mean, cov, 100000).T
+
+    fit = Fit(pdf, x=xdata, y=ydata, objective=LogLikelihood)
+    fit_result = fit.execute()
+
+    assert fit_result.value(x0) / mean[0] == pytest.approx(1, 1e-2)
+    assert fit_result.value(y0) / mean[1] == pytest.approx(1, 1e-2)
+    assert fit_result.value(
+        sig_x) / np.sqrt(cov[0, 0]) == pytest.approx(1, 1e-2)
+    assert fit_result.value(
+        sig_y) / np.sqrt(cov[1, 1]) == pytest.approx(1, 1e-2)
+    assert fit_result.value(rho) / r == pytest.approx(1, 1e-2)
+
+    marginal = integrate(pdf, (y, -oo, oo), conds='none')
+    fit = Fit(marginal, x=xdata, objective=LogLikelihood)
+    try:
+        # Should raise a NameError, not a TypeError, see #219
+        fit.execute()
+        assert False
+    except NameError:
+        assert True
+
+
+def test_evaluate_model():
+    """
+    Makes sure that models are callable and give the expected answer.
+    """
+    A = Parameter('A')
+    x = Variable('x')
+    new = A * x ** 2
+
+    assert new(x=2, A=2) == 8
+    assert not new(x=2, A=3) == 8
+
+
+def test_simple_sigma():
+    """
+    Make sure we produce the same results as scipy's curve_fit, with and
+    without sigmas, and compare the results of both to a known value.
+    """
+    t_data = np.array([1.4, 2.1, 2.6, 3.0, 3.3])
+    y_data = np.array([10, 20, 30, 40, 50])
+
+    sigma = 0.2
+    n = np.array([5, 3, 8, 15, 30])
+    sigma_t = sigma / np.sqrt(n)
+
+    # We now define our model
+    y = Variable('x')
+    g = Parameter('g')
+    t_model = (2 * y / g)**0.5
+
+    fit = Fit(t_model, y_data, t_data)  # , sigma=sigma_t)
+    fit_result = fit.execute()
+
+    # h_smooth = np.linspace(0,60,100)
+    # t_smooth = t_model(y=h_smooth, **fit_result.params)
+
+    # Lets with the results from curve_fit, no weights
+    popt_noweights, pcov_noweights = curve_fit(
+        lambda y, p: (2 * y / p)**0.5, y_data, t_data)
+
+    assert fit_result.value(g) == pytest.approx(popt_noweights[0])
+    assert fit_result.stdev(g) == pytest.approx(
+        np.sqrt(pcov_noweights[0, 0]), 1e-6)
```

Comment:
If you can do `approx(numpy_array)` it might be an idea to replace all `numpy.testing.assert_almost_equal` calls. Depending on your mood/time we can also leave it for later.

Reply:
I'll replace every `numpy.testing.assert_almost_equal` and `numpy.testing.assert_allclose` if two arguments are used (not three or more).

Reply:
How about `assert np.allclose(arr1, arr2)`?

Reply:
Pytest cannot compare in the test *model* actual models, but numpy. So I let it stand at two lines

Reply:
numpy can comare lists, pytest not